microsoft/testsuites/vm_extensions/azureperformancediagnostics.py

```diff
@@ -0,0 +1,186 @@
+        assert isinstance(platform, AzurePlatform)
+
+        # Create storage account and get credentials
+        random_str = generate_random_chars(string.ascii_lowercase + string.digits, 10)
+        storage_account_name = f"lisasc{random_str}"
+        information = environment.get_information()
+        resource_group_name = information["resource_group_name"]
+        location = information["location"]
+
+        check_or_create_storage_account(
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            resource_group_name=resource_group_name,
+            location=location,
+            log=log,
+        )
+
+        account_credential = get_storage_credential(
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            resource_group_name=resource_group_name,
+        )
+
+        # Run VM Extension
+        extension = node.features[AzureExtension]
+        settings = {
+            "performanceScenario": "quick",
+            "traceDurationInSeconds": 5,
+            "srNumber": "",
+            "requestTimeUtc": datetime.now(timezone.utc).strftime("%Y%m%d%H%M%S%f"),
+            "storageAccountName": storage_account_name,
+            "resourceId": (
+                f"/subscriptions/{platform.subscription_id}/resourceGroups/"
+                f"{resource_group_name}/providers/Microsoft.Compute/virtualMachines"
+                f"/{node.name}"
+            ),
+        }
+
+        protected_settings = {
+            "storageAccountName": storage_account_name,
+            "storageAccountKey": account_credential.get("account_key"),
+        }
+
+        extension_result = extension.create_or_update(
+            name="AzurePerformanceDiagnosticsLinux",
+            publisher="Microsoft.Azure.Performance.Diagnostics",
+            type_="AzurePerformanceDiagnosticsLinux",
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+            protected_settings=protected_settings,
+        )
+
+        assert_that(extension_result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Verify report was created and uploaded to the storage account
+        self._container_client = get_or_create_storage_container(
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            container_name="azdiagextnresults",
+            resource_group_name=resource_group_name,
+        )
+
+        blob_iter = self._container_client.list_blobs(
+            name_starts_with="PerformanceDiagnostics"
```

Comment:
how we know this blob is generated for the test run if there are many blobs with name prefix is `PerformanceDiagnostics`

Reply:
The storage account is spun up specifically for this test, so there shouldn't be other blobs in it, especially now that I've added a delete storage account method at your suggestion.